test/common/index.js

```diff
@@ -698,19 +698,24 @@ Object.defineProperty(exports, 'hasSmallICU', {
 });
 
 // Useful for testing expected internal/error objects
-exports.expectsError = function expectsError({code, type, message}) {
+exports.expectsError = function expectsError(options) {
   return function(error) {
-    assert.strictEqual(error.code, code);
-    if (type !== undefined) {
-      assert(error instanceof type,
-             `${error} is not the expected type ${type}`);
-    }
-    if (message instanceof RegExp) {
-      assert(message.test(error.message),
-             `${error.message} does not match ${message}`);
-    } else if (typeof message === 'string') {
-      assert.strictEqual(error.message, message);
-    }
+    Object.keys(options).forEach((key) => {
+      const actual = error[key];
+      const expected = options[key];
+      if (key === 'type') {
+        if (expected !== undefined) {
```

Comment:
That does not work as the class is passed to the function and not the instance. The simplest check I could though of is: `expected !== Error && !Error.isPrototypeOf(expected)`